app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,48 @@
+class CommentsController < ApplicationController
+
+  def new
+  	@riddle = Riddle.find_by_id(params[:riddle_id])
+  	@comment = Comment.new
+  end
+
+  def create
+  	new_comment = Comment.new(comment_params)
+  	if new_comment.save
```

Comment:
You could do this more railsy by doing 
params.require(:comment).permit(:body).merge(user_id: current_user.id)
in your comment_params.
Then these lines could collapse to riddle.comments.build(comment_params)
if riddle.save ...

The way this is now you are loading up objects from the db unnecessarily, just to get their ids, which you already know from the params.